pylds/base.py

```diff
@@ -186,18 +286,32 @@ def compute_lagrangian_descriptor(grid_parameters, vector_field, tau, p_value=0.
     LD : ndarray, shape (Nx, Ny)
         Array of computed Lagrangian descriptor values for all initial conditions.
     """
+    #get visualisation slice parameters and Number of DoF
+    if type(grid_parameters) == dict:
+        #n-DoF systems
+        slice_parameters = grid_parameters['slice_parameters'] # 2n-D grid
+        N_dof = len(grid_parameters['dof_slice'])
+    else:
+        #1-DoF systems
+        slice_parameters = grid_parameters # 2-D grid
+        N_dof = len(slice_parameters)
+        
+    #set boundaries for escape-box condition, if not defined
+    if not box_boundaries:
+        box_boundaries = int(N_dof/2)*[[-np.infty, np.infty]] #restricted to configuration space
     
-    f = lambda x, y: vector_field_flat(x, y, vector_field, p_value, box_boundaries)
-    y0 = generate_points(grid_parameters)
+    #solve initial value problem
+    f = lambda t, y: vector_field_flat(t, y, vector_field, p_value, box_boundaries)
+    y0, mask = generate_points(grid_parameters)
     
     solution = solve_ivp(f, [0,tau], y0, t_eval=[tau], rtol=1.0e-4)
+
+    LD_values = solution.y[N_dof::N_dof+1] #values corresponding to LD
```

Comment:
Is `N_dof+1` in `solution.y[N_dof::N_dof+1]` needed? `solution.y[N_dof]` should already be the last column corresponding to LDs

Reply:
Actually the above is correct. `solution.y` is a 1D array. So, the LD_values will be extracted by taking every subsequent `N_dof+1` item, as done above.

Reply:
Rigth, now I see, this way you avoided reshaping the array